### PR TITLE
Updated embeddings

### LINKED ISSUE #163: EhrEmbeddings does not require age/abspos as position_ids
`EhrEmbeddings` passes in `age` and `abspos` as `position_ids`, but now we explicitly pass in `input_embeds` to the model, so we don't need to wrap into `position_ids`

See [model](https://github.com/FGA-DIKU/EHR/blob/cceb21611e1a001ea3278c4cab2a97fd582b767e/corebehrt/modules/model/model.py) and [embeddings](https://github.com/FGA-DIKU/EHR/blob/main/corebehrt/modules/model/embeddings.py)

### LINKED ISSUE #166: Introduce `padding_idx` to `nn.Embedding`
Set the `padding_idx` in the `nn.Embedding` inside the `EhrEmbeddings` class which makes the padding_token a zero-vector. Might be redundant, but [ModernBERT does it](https://github.com/huggingface/transformers/blob/752ef3fd4e70869626ec70657a770a85c0ad9219/src/transformers/models/modernbert/modeling_modernbert.py#L202)

From https://pytorch.org/docs/stable/generated/torch.nn.Embedding.html: "If specified, the entries at `padding_idx` do not contribute to the gradient; therefore, the embedding vector at `padding_idx` is not updated during training, i.e. it remains as a fixed “pad”. For a newly constructed Embedding, the embedding vector at `padding_idx` will default to all zeros, but can be updated to another value to be used as the padding vector."

### PR DESCRIPTION
**Pull request by mikkelfo**
Closes #166 and closes #163, but is blocked by #168 (so `padding_idx` isn't set for `segment_embedding`)

### CODE REVIEW

File: corebehrt/modules/model/model.py [hunk 1]
```diff
@@ -14,18 +14,17 @@ def __init__(self, config):
...
             embedding_dropout=config.embedding_dropout,
+            pad_token_id=config.pad_token_id,
         )
 
     def forward(self, batch: dict):
-        input_ids = batch["concept"]
-        token_type_ids = batch["segment"]
-        attention_mask = torch.ones_like(input_ids)
-        position_ids = {key: batch[key] for key in ["age", "abspos"]}
+        attention_mask = torch.ones_like(batch["concept"])
```
Comment (kirilklein) [inline corebehrt/modules/model/model.py:21]: Will you look at fixing the attention mask and unpadding as part of this PR?

**Review by kirilklein — APPROVED** [review]
Looks godd

File: corebehrt/modules/model/model.py (hunk 1 above)
Comment (mikkelfo) [reply corebehrt/modules/model/model.py:21]: No, I left it for another issue #164 and have a different branch for it - Don't want to overflow this one
